diff --git a/build.gradle b/build.gradle
index 417383de..f4d784c2 100644
--- a/build.gradle
+++ b/build.gradle
@@ -433,6 +433,7 @@ project(':core') {
     compile libs.slf4jlog4j
     compile libs.zkclient
     compile libs.zookeeper
+    compile libs.apacheCommonsLang
     // These modules were broken out of core scala in 2.10. We can remove special handling when 2.10 support is dropped.
     if (versions.baseScala != '2.10') {
       compile libs.scalaParserCombinators
@@ -629,6 +630,7 @@ project(':clients') {
   archivesBaseName = "kafka-clients"
 
   dependencies {
+    compile libs.apacheCommonsLang
     compile libs.lz4
     compile libs.snappy
     compile libs.slf4jApi
diff --git a/gradle/dependencies.gradle b/gradle/dependencies.gradle
index 936070a4..74236633 100644
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -23,6 +23,7 @@ ext {
 }
 
 versions += [
+  apachecommonslang: "3.5",
   apacheda: "1.0.0-M33",
   apacheds: "2.0.0-M21",
   argparse4j: "0.7.0",
@@ -71,6 +72,7 @@ if (hasProperty('scalaVersion')) {
 versions["baseScala"] = versions.scala.substring(0, versions.scala.lastIndexOf("."))
 
 libs += [
+  apacheCommonsLang: "org.apache.commons:commons-lang3:$versions.apachecommonslang",
   argparse4j: "net.sourceforge.argparse4j:argparse4j:$versions.argparse4j",
   apacheda: "org.apache.directory.api:api-all:$versions.apacheda",
   apachedsCoreApi: "org.apache.directory.server:apacheds-core-api:$versions.apacheds",
